### PR TITLE
Cleanup keystone v3 attribute and enable multi-domain support

### PR DESCRIPTION
**Pull request by rhafer**

### CODE REVIEW

**Comment by dirkmueller** [comment]
+1

File: chef/cookbooks/nova_dashboard/recipes/server.rb [hunk 1]
```diff
@@ -297,8 +277,7 @@
     :memcached_locations => memcached_locations,
     :can_set_mount_point => node["nova_dashboard"]["can_set_mount_point"],
     :can_set_password => node["nova_dashboard"]["can_set_password"],
-    :keystone_api_version => keystone_api_version,
-    :keystone_internal_auth_url => keystone_internal_auth_url,
+    :multi_domain_support => node["nova_dashboard"]["multi_domain_support"],
```
Comment (vuntz) [inline chef/cookbooks/nova_dashboard/recipes/server.rb]: Shouldn't we override this to False if keystone default API is not v3?

Comment (rhafer) [inline chef/cookbooks/nova_dashboard/recipes/server.rb]: ```validate_proposal_after_save``` doesn't allow to enable multi domain support, when keystone API version is set to 2.0. Do you think that's not enough?

Comment (vuntz) [inline chef/cookbooks/nova_dashboard/recipes/server.rb]: I missed that. But I guess I could argue that the user could go back and change keystone to 2.0 afterwards; which would result in the bug. Kind of a edge case, but I'd still protect against it.

**Comment by rhafer** [comment]
> Kind of a edge case, but I'd still protect against it.

Done

**Comment by mjura** [comment]
+1 LGTM

File: chef/data_bags/crowbar/migrate/nova_dashboard/033_remove_use_keystone_v3.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def upgrade ta, td, a, d
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/nova_dashboard/033_remove_use_keystone_v3.rb]: For the record, it's fine to have multiple scripts with the same revision.

File: chef/data_bags/crowbar/bc-template-nova_dashboard.json [hunk 3]
```diff
@@ -32,7 +32,7 @@
       },
       "can_set_mount_point": false,
       "can_set_password": false,
-      "use_keystone_v3": false,
+      "multi_domain_support": true,
```
Comment (toabctl) [inline chef/data_bags/crowbar/bc-template-nova_dashboard.json]: As long as we don't have keystone V3 as default, `multi_domain_support` should be `false`

**Comment by rsalevsky** [comment]
@rhafer, this repository got merged into [crowbar-openstack](https://github.com/crowbar/crowbar-openstack).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by rhafer** [comment]
Forward ported to new repository: https://github.com/crowbar/crowbar-openstack/issues/26

**Pull request closed without merging**